Marlin/src/feature/meatpack.h

```diff
@@ -146,31 +146,30 @@ struct MeatpackSerial : public SerialBase <MeatpackSerial < SerialT >> {
     // There is a potential issue here with multiserial, since it'll return its decoded buffer whatever the serial index here.
     // So, instead of doing MeatpackSerial<MultiSerial<...>> we should do MultiSerial<MeatpackSerial<...>, MeatpackSerial<...>>
     // TODO, let's fix this later on
-    if (charCount > 0) return charCount;
-
-    // Don't read in read method, instead do it here, so we can make progress in the read method
-    while (out.available(index) > 0) {
-      int r = out.read(index);
-      if (r == -1) return 0; // This is an error from the underlying serial code
-
-      meatpack.handle_rx_char((uint8_t)r, index);
-      charCount = meatpack.get_result_char(serialBuffer);
-      readIndex = 0;
-      if (charCount) return charCount;
+    if (!charCount) {
+      // Don't read in read method, instead do it here, so we can make progress in the read method
+      while (out.available(index) > 0) {
```

Comment:
Maybe this should be `while (!charCount && out.available(index))` to be equivalent to the old code at `if (charCount) return charCount;`

Reply:
I'm not sure. In the former case, if charCount is zero, we enter the loop, but can exit with charCount == 0 (if there's no byte available). In your example, you'd never exit in this situation, it can loop forever (potential loop lock) if meatpack never succeed decoding a char. I can't really tell since I haven't written Meatpack's code here if it's possible or not.